closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -685,13 +685,8 @@ private void initProviders() {
                 // Provider with argument (provider name + optional argument).
                 providers.add(pNum - 1, providerName);
 
-                // Remove the provider's optional arguments if present.
-                pos = providerName.indexOf(' ');
-                providerName = (pos < 0) ? providerName.trim() : providerName.substring(0, pos).trim();
-                // Remove the provider's class package names if present.
-                pos = providerName.lastIndexOf('.');
-                providerName = (pos < 0) ? providerName : providerName.substring(pos + 1, providerName.length());
-                // Provider without arguments and package names.
+                // Provider name defined in provider construction method.
+                providerName = getProvidersSimpleName(providerName);
```

Comment:
Previously, we did not adding a new method for trimming the provider name when there are packages or arguments. But when we back port the Restricted Security mode to JDK8, some providers name(name defined in the construction method) and its class name in JDK8 are not same, so only from the class name, we can not get the provider name. Then, we add the new method(getProvidersSimpleName) for mapping the provider name and its class name. And during those code reviews on JDK8, you suggested, we can move those provider name trimming codes into this new method. So, that is why, those trimming codes and provider name, class name mapping codes, are in this new method. 

But previously, these codes are only exist in JDK8. For this PR, we have to add the provider name and its class name mapping for provider SunSASL. So, I copy those same codes from JDK8 to this JDKNext PR, and also adding the mapping for SunSASL.

Reply:
I'm not sure you actually answered my question. I now think it was an oversight that the code did not previously trim the argument to form the simple name. Please confirm.

Reply:
I wouldn’t say it was an oversight that the code did not previously trim the argument (after -). Because, in the `initProviders()` method, it didn’t have the argument trim for `“-”`, that is because no need to trim `“-”`. The provider lists which the initProviders() method will deal with are as below:

```
RestrictedSecurity.NSS.140-2.jce.provider.1 = SunPKCS11 ${java.home}/conf/security/nss.fips.cfg
RestrictedSecurity.NSS.140-2.jce.provider.2 = SUN [{CertificateFactory, X.509, ImplementedIn=Software}, \
    {CertStore, Collection, ImplementedIn=Software}, \
    {CertStore, com.sun.security.IndexedCollection, ImplementedIn=Software}, \
    {Policy, JavaPolicy, *}, {Configuration, JavaLoginConfig, *}, \
    {CertPathBuilder, PKIX, ValidationAlgorithm=RFC5280:ImplementedIn=Software}, \
    {CertPathValidator, PKIX, ValidationAlgorithm=RFC5280:ImplementedIn=Software}, \
    {KeyStore, PKCS12, *}]
RestrictedSecurity.NSS.140-2.jce.provider.3 = SunEC [{KeyFactory, EC, ImplementedIn=Software: \
    SupportedKeyClasses=java.security.interfaces.ECPublicKey|java.security.interfaces.ECPrivateKey: \
    KeySize=256}, {AlgorithmParameters, EC, *}]
RestrictedSecurity.NSS.140-2.jce.provider.4 = SunJSSE
```

The provider `SunPKCS11`, before it initialized using the configure file ${java.home}/conf/security/nss.fips.cfg. Its name is `SunPKCS11`, not `SunPKCS11-NSS-FIPS` yet.